internal/pkg/api/router.go

```diff
@@ -104,8 +121,120 @@ func NewRouter(ctx context.Context, bulker bulk.Bulk, ct *CheckinT, et *Enroller
 			logger.HTTPHandler(handler),
 		)
 	}
+	log.Info().Str("addr", addr).Msg("fleet-server routes set up")
+	return router
+}
 
-	log.Info().Msg("fleet-server routes set up")
+// Run starts the api server on the listeners configured in the config.
+// Each listener has a unique limit.Limiter to allow for non-global rate limits.
+func (rt *Router) Run(ctx context.Context) error {
+	rt.ctx = ctx
 
-	return router
+	listeners := rt.cfg.BindEndpoints()
+	rdto := rt.cfg.Timeouts.Read
+	wrto := rt.cfg.Timeouts.Write
+	idle := rt.cfg.Timeouts.Idle
+	rdhr := rt.cfg.Timeouts.ReadHeader
+	mhbz := rt.cfg.Limits.MaxHeaderByteSize
+	bctx := func(net.Listener) context.Context { return ctx }
+
+	errChan := make(chan error)
+	cancelCtx, cancel := context.WithCancel(ctx) // TODO should we set rt.ctx = cancelCtx?
```

Comment:
No.  The cancel context wraps the lifetime of the server.  We are using the global context here in handleEnroll to do a rollback.  Perhaps clearer to rename ctx in the router to "baseCtx" or "appCtx".